app/lib/package.rb

```diff
@@ -56,20 +64,25 @@ def add_pages(zip_file, manifest)
 
   def write_page_to_file(page)
     translated_page = @translation.translated_page(page.id, true)
+    document = Nokogiri::XML(translated_page)
+    determine_resources(document)
     sha_filename = XmlUtil.xml_filename_sha(translated_page)
 
     File.write("#{@directory}/#{sha_filename}", translated_page)
 
     sha_filename
   end
 
-  def add_attachments(zip_file, manifest)
-    @translation.resource.attachments.where(is_zipped: true).each do |a|
-      Rails.logger.info("Adding attachment with id: #{a.id} to package for translation with id: #{@translation.id}")
+  def add_attachments(zip_file, manifest) # rubocop:disable Metrics/AbcSize
+    @resources.uniq.each do |filename|
+      attachment = @translation.resource.attachments.find_by(file_file_name: filename)
```

Comment:
Can there be multiple attachments with the same `file_file_name`? The prior `.where` would find them but `.find_by` will not.

Reply:
There shouldn't be, and if there are the apps wouldn't know what to do with multiple files with the same name anyways